src/index.d.ts

```diff
@@ -84,24 +84,25 @@ declare namespace preact {
 	// Type alias for a component instance considered generally, whether stateless or stateful.
 	type AnyComponent<P = {}, S = {}> = FunctionComponent<P> | Component<P, S>;
 
-	interface Component<P = {}, S = {}> {
+	interface Component<P = {}, S = {}, C = any> {
 		componentWillMount?(): void;
 		componentDidMount?(): void;
 		componentWillUnmount?(): void;
 		getChildContext?(): object;
-		componentWillReceiveProps?(nextProps: Readonly<P>, nextContext: any): void;
-		shouldComponentUpdate?(nextProps: Readonly<P>, nextState: Readonly<S>, nextContext: any): boolean;
-		componentWillUpdate?(nextProps: Readonly<P>, nextState: Readonly<S>, nextContext: any): void;
+		componentWillReceiveProps?(nextProps: Readonly<P>, nextContext: C): void;
```

Comment:
Thinking we should mark context as Readonly here like props and state are treated.

Reply:
Makes sense, thanks!